src/components/TimeseriesPreview/TimeseriesPreview.tsx

```diff
@@ -0,0 +1,329 @@
+  const [timeseries, setTimeseries] = useState<GetTimeSeriesMetadataDTO>();
+  const [intervalPointer, setIntervalPointer] = useState<number | undefined>();
+  const [latestDatapoint, setLatestDatapoint] = useState<
+    GetDoubleDatapoint | GetStringDatapoint
+  >();
+
+  const onVisibilityClick = () => {
+    if (toggleVisibility && timeseries) {
+      toggleVisibility(timeseries);
+    }
+  };
+
+  const dropdownMenu =
+    dropdown && timeseries ? (
+      <Dropdown
+        overlay={generateDropdownMenu({
+          ...dropdown,
+          timeseries,
+          styles: dropdownStyle,
+        })}
+        trigger={['click']}
+      >
+        <ActionIcon type="ellipsis" />
+      </Dropdown>
+    ) : null;
+
+  const displayValue = () => {
+    const value = valueToDisplay
+      ? valueToDisplay.value
+      : latestDatapoint
+      ? latestDatapoint.value
+      : undefined;
+
+    return formatDisplayValue ? formatDisplayValue(value) : value;
+  };
+
+  const displayDate = () => {
+    const date = valueToDisplay
+      ? valueToDisplay.timestamp
+      : latestDatapoint
+      ? latestDatapoint.timestamp
+      : undefined;
+
+    return date ? moment(date).format(dateFormat) : undefined;
+  };
+
+  useEffect(() => {
+    let canceled = false;
+
+    const fetchTimeseries = async (id: number): Promise<void> => {
+      try {
+        const timeseriesById = retrieveTimeseries
+          ? await retrieveTimeseries({ id })
+          : await cachedContext!.timeseries.retrieve([{ id }]);
+
+        if (timeseriesById.length && !canceled) {
+          setTimeseries(timeseriesById[0]);
+        }
+      } catch (e) {
+        console.error(e);
+      }
+
+      setLoading(false);
+    };
+
+    fetchTimeseries(timeseriesId);
+
+    return () => {
+      canceled = true;
+    };
+  }, [timeseriesId]);
+
+  useEffect(() => {
+    let canceled = false;
+
+    const fetchLatestDatapoint = async (id: number): Promise<void> => {
+      try {
+        const latest = retrieveLatestDatapoint
+          ? await retrieveLatestDatapoint({ id })
+          : await context!.datapoints.retrieveLatest([{ id }]);
+
+        if (latest.length && latest[0].datapoints[0] && !canceled) {
+          setLatestDatapoint(latest[0].datapoints[0]);
+        }
+      } catch (e) {
+        console.error(e);
+      }
+    };
+
+    clearInterval(intervalPointer);
+
+    if (typeof valueToDisplay === 'undefined') {
+      const pointer = setInterval(() => {
+        if (canceled) {
+          clearInterval(intervalPointer);
+
+          return;
+        }
+
+        fetchLatestDatapoint(timeseriesId);
+      }, updateInterval);
+
+      fetchLatestDatapoint(timeseriesId);
+
+      setIntervalPointer(pointer);
+    }
+
+    return () => {
+      canceled = true;
+    };
+  }, [valueToDisplay, timeseriesId]);
+
+  return (
+    <Wrapper style={wrapperStyle}>
+      <Card
+        bordered={false}
+        bodyStyle={{ padding: 0, ...cardStyle }}
+        loading={loading}
```

Comment:
Idk the reason but skeleton that shows-up on loading has different heigh then the actual component (it jumps when timeseries is loaded)

Reply:
cause its placeholder that has static height, and we can't know real height of component until it will be loaded